### PR TITLE
added search feature

### PR DESCRIPTION
**Pull request by Laaidback**

### CODE REVIEW

File: app/src/main/java/com/example/krysztian/notes/ui/activities/MainActivity.java [hunk 1]
```diff
@@ -66,8 +72,23 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
 
     @Override
     public boolean onCreateOptionsMenu(Menu menu) {
-        MenuInflater menuInflater = getMenuInflater();
-        menuInflater.inflate(R.menu.activity_main_actions, menu);
+       getMenuInflater().inflate(R.menu.activity_main_actions, menu);
+        final SearchView searchView = (SearchView) MenuItemCompat.getActionView(menu.findItem(R.id.action_search));
+        SearchManager searchManager = (SearchManager) getSystemService(SEARCH_SERVICE);
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
```
Comment (duchuyctlk) [inline app/src/main/java/com/example/krysztian/notes/ui/activities/MainActivity.java]: If you really want to perform searching whenever a TextChange event fires, please consider using RxAndroid to reduce the amount of consecutive search requests.

File: app/src/main/java/com/example/krysztian/notes/ui/activities/MainActivity.java [hunk 2]
```diff
@@ -133,12 +159,42 @@ private void getNotes() {
...
+
+        for (int i = src.length() - length; i >= 0; i--) {
+            final char ch = src.charAt(i);
+            if (ch != firstLo && ch != firstUp) continue;
+            if (src.regionMatches(true, i, what, 0, length)) return true;
+        }
+        return false;
+    }
+
 }
```
Comment (duchuyctlk) [inline app/src/main/java/com/example/krysztian/notes/ui/activities/MainActivity.java]: Please consider adding an empty line at the end of each source file to make GitHub happy 😄 .

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -6,55 +6,81 @@
...
-    <LinearLayout
-        android:id="@+id/mainLinearLayout"
+    <RelativeLayout
         android:layout_width="match_parent"
         android:layout_height="match_parent"
         android:background="@color/colorAccent"
-        android:orientation="vertical"
-        android:padding="15dp"
-        tools:context="com.example.krysztian.notes.ui.activities.MainActivity">
+        android:padding="15dp">
```
Comment (duchuyctlk) [inline app/src/main/res/layout/activity_main.xml]: Dimensions like this should be stored in `dimens.xml` in `res` folder.

File: app/src/main/java/com/example/krysztian/notes/ui/activities/MainActivity.java [hunk 4]
```diff
@@ -66,8 +72,23 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
 
     @Override
     public boolean onCreateOptionsMenu(Menu menu) {
-        MenuInflater menuInflater = getMenuInflater();
-        menuInflater.inflate(R.menu.activity_main_actions, menu);
+       getMenuInflater().inflate(R.menu.activity_main_actions, menu);
+        final SearchView searchView = (SearchView) MenuItemCompat.getActionView(menu.findItem(R.id.action_search));
```
Comment (navarionek) [inline app/src/main/java/com/example/krysztian/notes/ui/activities/MainActivity.java]: indentation of this line and above one doesn't match

File: app/src/main/java/com/example/krysztian/notes/ui/activities/MainActivity.java [hunk 5]
```diff
@@ -133,12 +159,42 @@ private void getNotes() {
...
+        List<Note> searchResults = new ArrayList<>();
+        for (Note note : notes) {
+            if (containsIgnoreCase(note.getTitle(), query) || containsIgnoreCase(note.getText(),query)) {
+                searchResults.add(note);
+            }
+        }
+        adapter.setNotes(searchResults);
+    }
+
+    private boolean containsIgnoreCase(String src, String what) {
```
Comment (navarionek) [inline app/src/main/java/com/example/krysztian/notes/ui/activities/MainActivity.java]: use src.toLowerCase().contains(what.toLowerCase());

These kinds of methods are almost often ready to use in frameworks

**Pull request merged**